File: Electronics_Shop_17.Model/SearchObjects/SearchCoupon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.SearchObjects
{
    public class SearchCoupon : BaseSearch
    {
        public int? Id { get; set; }
        public string? Code { get; set; }
        public int? DiscountAmount { get; set; }
        public int? MinPurchaseAmount { get; set; }
        public int? MaxUsagePerCustomer { get; set; }
        public int? ProductCategoryId { get; set; }
        public int? AccessoryCategoryId { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool? IsActive { get; set; }

        public bool? isDeleted { get; set; }
    }
}

File: angular17_web/src/app/models/coupon/search.coupon.ts
import { BaseSearch } from "../base.search";

export interface SearchCoupon extends BaseSearch {
  id?: number;
  code?: string;
  discountAmount?: number;
  minPurchaseAmount?: number;
  maxUsagePerCustomer?: number;
  productCategoryId?: number;
  accessoryCategoryId?: number;
  startDate?: Date;
  endDate?: Date;
  isActive?: boolean;
  isDeleted?: boolean;
}
